server/aesdsocket.c

```diff
@@ -0,0 +1,332 @@
+/*
+* file: aesdsocket.c
+* 
+* purpose: opens a socket @ port 9000, logs message to syslog upon connection, receives data over connection to file, *	returns full content of file to client
+*	logs closed connection
+*	restarts accepting connection until SIGINT or SIGTERM and gracefully exits
+*	deletes the temp file
+*
+* author: Chris Choi
+*
+* date: 03 October 2021
+*
+*/
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <arpa/inet.h> 
+#include <fcntl.h>  
+#include <signal.h> 
+#include <syslog.h> 
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/wait.h>
+#include <netdb.h>
+#include <getopt.h>
+#include <netinet/in.h>
+
+
+#define	PORT			    "9000"
+#define	FILE_OUT_PATH		"/var/tmp/aesdsocketdata"
+#define	BUFFER_SIZE		    100
+#define	MAX_CONNECTION		4
+
+#define FD_SIZE     3
+#define FD_DATA     0
+#define FD_CLIENT   1
+#define FD_SOCKET   2
+
+#define SUCCESS     0
+
+int daemonFlag = 0;
+
+char * buffer_rx;
+char * buffer_tx;
+
+struct sockaddr_in    server_address;
+struct sockaddr_in    client_address;
+
+int fd[FD_SIZE];
+
+pid_t pid;
+
+
+static void sighandler(int signum)
+{
+	if((signum == SIGTERM) || signum == SIGINT)
+    {
+
+        syslog(LOG_DEBUG, "SIGINT or SIGTERM received, stopping...\n");
+
+        for(int i = 0; i < FD_SIZE; i++)
+        {
+            close(fd[i]);
+        }
+
+        free(buffer_rx);
+        free(buffer_tx);
+
+        closelog();
+        remove(FILE_OUT_PATH);
+
+        exit(SUCCESS);
```

Comment:
@cchoi22915 consider a design where you don't need to exit() from a signal handler.